src/scripts/main.js

```diff
@@ -1,5 +1,36 @@
 'use strict';
 
-document.addEventListener('click', e => {
-  // write code here
+const wall = document.querySelector('.wall');
+const spider = document.querySelector('.spider');
+
+wall.addEventListener('click', e => {
+  const wallCoords = wall.getBoundingClientRect();
+
+  const spiderCoords = {
+    top:
+      e.clientY - wallCoords.top - wall.clientTop
+      - spider.clientHeight / 2,
+    left:
+      e.clientX - wallCoords.left - wall.clientLeft
+      - spider.clientWidth / 2,
+  };
+
+  if (spiderCoords.left < 0) {
+    spiderCoords.left = 0;
+  }
+
+  if (spiderCoords.top < 0) {
+    spiderCoords.top = 0;
+  }
+
+  if (spiderCoords.top + spider.clientHeight > wall.clientHeight) {
+    spiderCoords.top = wall.clientHeight - spider.clientHeight;
+  }
+
+  if (spiderCoords.left + spider.clientWidth > wall.clientWidth) {
+    spiderCoords.left = wall.clientWidth - spider.clientWidth;
+  }
```

Comment:
let's simplify the code and create ONE separate function, which will return the valid coords